lib/group_repository.py

```diff
@@ -25,6 +28,60 @@
 logger = get_logger(__name__)
 
 
+def add_hosts_to_group(group_id: str, host_id_list: List[str]):
+    current_org_id = get_current_identity().org_id
+
+    # Check if the hosts are already associated with another group
+    assoc_query = HostGroupAssoc.query.filter(HostGroupAssoc.host_id.in_(host_id_list)).all()
+    if assoc_query:
+        taken_hosts = [assoc.host_id for assoc in assoc_query]
+        log_host_group_add_failed(logger, taken_hosts, group_id)
```

Comment:
```suggestion
        log_host_group_add_failed(logger, host_id_list, group_id)
```
I'm not sure about this, so feel free to ignore this suggestion if you think we should only log the bad hosts.